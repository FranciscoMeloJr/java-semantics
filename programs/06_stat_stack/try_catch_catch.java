// Testing try with multiple catch clauses

class Exception2 extends RuntimeException {
  void Exception2() {
  }
}

class Exception3 extends Exception2 {
  void Exception3() {
  }
}

public class try_catch_catch {
  public static void main(String[] args) {
    try {
      if (true) throw new Exception2();

      System.out.println("No exception");
    } catch (Exception3 e3) {
      System.out.println("Caught Exception3");
    } catch (Exception2 e2) {
      System.out.println("Caught Exception2");
    } catch (RuntimeException e) {
      System.out.println("Caught RuntimeException");
    }
    System.out.println("Done!");
  }
}

// Caught Exception2
// Done!
