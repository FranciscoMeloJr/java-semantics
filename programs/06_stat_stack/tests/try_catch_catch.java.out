Caught exception2
Done!
